axelor-contract/src/main/java/com/axelor/apps/contract/batch/BatchContractRevaluate.java

```diff
@@ -0,0 +1,218 @@
+      processContract(idsFail, idsOk, idsReevaluated, contract);
+    }
+    return Stream.of(
+            new AbstractMap.SimpleEntry<>("OK", idsOk),
+            new AbstractMap.SimpleEntry<>("FAIL", idsFail),
+            new AbstractMap.SimpleEntry<>("REEVALUATED", idsReevaluated))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  @Transactional(rollbackOn = Exception.class)
+  protected void processContract(
+      List<Contract> idsFail,
+      List<Contract> idsOk,
+      List<Contract> idsReevaluated,
+      Contract contract) {
+    try {
+      contract = contractRepository.find(contract.getId());
+      Contract newContract = contractService.getNextContract(contract);
+      processContract(newContract, idsOk, idsReevaluated);
+    } catch (Exception e) {
+      TraceBackService.trace(e, null, batch.getId());
+      idsFail.add(contract);
+      incrementAnomaly();
+    }
+  }
+
+  void processContract(Contract contract, List<Contract> idsOk, List<Contract> idsReevaluated)
+      throws AxelorException, ScriptException {
+    LocalDate todayDate = appBaseService.getTodayDate(contract.getCompany());
+    LocalDate nextRevaluationDate = contract.getNextRevaluationDate();
+    LocalDate lastRevaluationDate = contract.getLastRevaluationDate();
+
+    if (launchBatch(todayDate, contract)) {
+      contract = contractRevaluationService.applyFormula(contract);
+      contract.setLastRevaluationDate(todayDate);
+      contract.setNextRevaluationDate(computeNextRevaluationDate(contract, todayDate));
+
+      idsReevaluated.add(contract);
+      incrementDone();
+    } else if (nextRevaluationDate == null && lastRevaluationDate != null) {
+      contract.setNextRevaluationDate(
+          computeNextRevaluationDate(contract, lastRevaluationDate).minusDays(5));
+    } else if (nextRevaluationDate == null) {
+      contract.setNextRevaluationDate(computeNextRevaluationDate(contract, null).minusDays(5));
+    }
+    idsOk.add(contract);
+    contractRepository.save(contract);
+  }
+
+  protected boolean launchBatch(LocalDate todayDate, Contract contract) {
+    LocalDate nextRevaluationDate = contract.getNextRevaluationDate();
+    LocalDate lastRevaluationDate = contract.getLastRevaluationDate();
+    LocalDate startDate = contract.getStartDate();
+    Duration revaluationPeriod = contract.getRevaluationPeriod();
+    LocalDate computedLastRevaluationDate =
+        computeNewDateWithPeriod(revaluationPeriod, lastRevaluationDate);
+    LocalDate computedStartDate = computeNewDateWithPeriod(revaluationPeriod, startDate);
+
+    return nextRevaluationDate != null
+        && (todayDate.isEqual(nextRevaluationDate)
+            || (nextRevaluationDate.isBefore(todayDate)
+                && ((computedLastRevaluationDate != null
+                        && computedLastRevaluationDate.isBefore(nextRevaluationDate))
+                    || (computedStartDate != null && computedStartDate.isBefore(todayDate)))));
```

Comment:
I'm not sure about this condition, the accumulation of `&&` and `||` could lead to future mistakes, can you check a way to make it more readable ?